retrofit/src/main/java/retrofit2/RequestBuilder.java

```diff
@@ -107,19 +106,16 @@ private static String canonicalize(String input, boolean alreadyEncoded) {
     return input;
   }
 
-  private static void canonicalize(Buffer out, String input, int pos, int limit,
-      boolean alreadyEncoded) {
+  private static void canonicalizeForPath(Buffer out, String input, int pos, int limit,
+                                          boolean alreadyEncoded) {
     Buffer utf8Buffer = null; // Lazily allocated.
     int codePoint;
     for (int i = pos; i < limit; i += Character.charCount(codePoint)) {
       codePoint = input.codePointAt(i);
       if (alreadyEncoded
-          && (codePoint == '\t' || codePoint == '\n' || codePoint == '\f' || codePoint == '\r')) {
+          && shouldSkipCodePoint(codePoint)) {
         // Skip this character.
-      } else if (codePoint < 0x20
-          || codePoint >= 0x7f
-          || PATH_SEGMENT_ENCODE_SET.indexOf(codePoint) != -1
-          || (codePoint == '%' && !alreadyEncoded)) {
+      } else if (!alreadyEncoded && shouldEncodeCodePoint(codePoint)) {
```

Comment:
Thanks, but not sure if i understand. If the API creator says its encoded, why should if be re-encoded? If the caller passed in a bad value with the understanding it was to be encoded, why would they have any expectation it would work?